apps/ui/src/hooks/Communities/useUserCommunitiesStatus.tsx

```diff
@@ -0,0 +1,21 @@
+import { UserStatus } from '@communecar/types';
+import { getRandomOption } from '../../utils';
+import { useGetAllCommunities } from './useGetAllCommunities';
+import { UserCommunitiesStatus } from '../../types/community-type';
+
+const userStatusOptions: UserStatus[] = ['Approved', 'Pending', 'Rejected'];
+
+const useUserCommunitiesStatus = (userId: string): UserCommunitiesStatus => {
+  const communities = useGetAllCommunities();
+  console.log({userId}); //TODO when the server ready
+  
+
+  const communitiesStatus = communities.reduce((acc, community) => {
```

Comment:
It is better to not use reduce - prefer iteration with abject initialization (a = [], forEach...)
- more clear to read
- better performance in all cases